girder/api/v1/folder.py

```diff
@@ -82,10 +86,41 @@ def find(self, parentType, parentId, text, name, limit, offset, sort):
                 offset=offset, limit=limit, sort=sort, filters=filters)
         elif text:
             return self._model.textSearch(
-                text, user=user, limit=limit, offset=offset, sort=sort)
+                text, user=user, limit=limit, offset=offset, sort=sort, filters=filters)
         else:
             raise RestException('Invalid search mode.')
 
+    @access.public(scope=TokenScope.DATA_READ)
+    @autoDescribeRoute(
+        Description('Report the offset of a folder in a list or search.')
+        .notes('You must pass either a "folderId" or "text" field '
+               'to specify how you are searching for folders.  '
+               'If you omit one of these parameters the request will fail and respond : '
+               '"Invalid search mode."')
+        .modelParam('id', model=FolderModel, level=AccessType.READ)
+        .param('parentType', "Type of the folder's parent", required=False,
+               enum=['folder', 'user', 'collection'])
+        .param('parentId', "The ID of the folder's parent.", required=False)
+        .param('text', 'Pass to perform a text search.', required=False)
+        .param('name', 'Pass to lookup a folder by exact name match. Must '
+               'pass parentType and parentId as well when using this.', required=False)
+        .pagingParams(defaultSort='lowerName')
+        .errorResponse()
+        .errorResponse('Read access was denied on the parent resource.', 403)
+    )
+    def findPosition(self, folder, parentType, parentId, text, name, limit, offset, sort):
+        filters = {}
+        if len(sort) != 1 or sort[0][0] not in folder:
+            raise RestException('Invalid sort mode.')
+        dir = '$lt' if sort[0][1] == SortDir.ASCENDING else '$gt'
+        filters = {'$or': [
+            {sort[0][0]: {dir: folder.get(sort[0][0])}},
+            {sort[0][0]: folder.get(sort[0][0]), '_id': {dir: folder['_id']}}
```

Comment:
I'm having trouble understanding the second condition of the query. Why are we doing a `$gt` or `$lt` comparison on the `_id` field? Are you just trying to check `$ne`?

Reply:
There may be duplicates on the sort value, and I would like to return an expected answer.  For instance, you have a list `a, b, b, b, c`, and we ask for a folder with name `b`, the position could be 1, 2, or 3.  But, in this instance, I sort by the `_id` within that so the three identically named folders return distinct positions.

We try to prevent duplicate names (but not very hard, so they can still happen).  Further, you could sort by something other than name (size, for instance) that can naturally be duplicated.  Ideally, all of our sorted results from find would have a fall-back sort, so a sort-by-name would really be sort-by-name-then-by-_id, and this would match.  I'm using _id as the final component, because mongo seems to (but this isn't guaranteed in the docs as far as I can tell).

Reply:
Oh I see, you're assuming that mongodb sorting is stable and ordered by `_id`. Could you add a comment to explain the logic? And include that this looks like we're depending on undefined behavior?